src/vm/i386/asmhelpers.S

```diff
@@ -472,10 +472,12 @@ NESTED_END CallDescrWorkerInternal, _TEXT
 #ifdef _DEBUG
 // int __fastcall HelperMethodFrameRestoreState(HelperMethodFrame*, struct MachState *)
 LEAF_ENTRY HelperMethodFrameRestoreState, _TEXT
+    #define     RETN 8
```

Comment:
The @4 doesn't mean it would clean up the stack. It means the size of the arguments. See this web page:
https://msdn.microsoft.com/en-us/library/56h2zst2.aspx
>`__fastcall	Leading and trailing at signs (@) followed by a decimal number representing the number of bytes in the parameter list`

I have tried to compile a simple piece of code using clang with the -m32 option and a function marked with __fastcall. There was no extra parameter on the ret:
```C++
struct LazyMachState
{
  int x;
};

int __fastcall LazyMachStateCaptureState(struct LazyMachState *pState)
{
  return 1;
}
```
```asm
        .globl  LazyMachStateCaptureState(LazyMachState*)
        .p2align        4, 0x90
        .type   LazyMachStateCaptureState(LazyMachState*),@function
LazyMachStateCaptureState(LazyMachState*): # @LazyMachStateCaptureState(LazyMachState*)
.Lfunc_begin0:
        .loc    1 13 0                  # /tmp/gcc-explorer-compiler1161112-73-oxuub7/example.cpp:13:0
        .cfi_startproc
        push    ebp
        .cfi_def_cfa_offset 8
        .cfi_offset ebp, -8
        mov     ebp, esp
        .cfi_def_cfa_register ebp
        push    eax
        mov     eax, 1
        mov     dword ptr [ebp - 4], ecx
        .loc    1 14 3 prologue_end     # /tmp/gcc-explorer-compiler1161112-73-oxuub7/example.cpp:14:3
        add     esp, 4
        pop     ebp
        ret
.Lfunc_end0:
        .size   LazyMachStateCaptureState(LazyMachState*), .Lfunc_end0-LazyMachStateCaptureState(LazyMachState*)
        .cfi_endproc
```